datafusion/common/src/table_reference.rs

```diff
@@ -155,50 +178,110 @@ impl<'a> TableReference<'a> {
                 table,
             },
             Self::Partial { schema, table } => ResolvedTableReference {
-                catalog: default_catalog,
+                catalog: default_catalog.into(),
                 schema,
                 table,
             },
             Self::Bare { table } => ResolvedTableReference {
-                catalog: default_catalog,
-                schema: default_schema,
+                catalog: default_catalog.into(),
+                schema: default_schema.into(),
                 table,
             },
         }
     }
 
-    /// Forms a [`TableReference`] by splitting `s` on periods `.`.
-    ///
-    /// Note that this function does NOT handle periods or name
-    /// normalization correctly (e.g. `"foo.bar"` will be parsed as
-    /// `"foo`.`bar"`. and `Foo` will be parsed as `Foo` (not `foo`).
-    ///
-    /// If you need to handle such identifiers correctly, you should
-    /// use a SQL parser or form the [`OwnedTableReference`] directly.
+    /// Forms a [`TableReference`] by attempting to parse `s` as a multipart identifier,
+    /// failing that then taking the entire input as the identifier itself.
     ///
-    /// See more detail in <https://github.com/apache/arrow-datafusion/issues/4532>
+    /// Will normalize (convert to lowercase) any unquoted identifiers.
+    /// e.g. `Foo` will be parsed as `foo`, and `"Foo"".bar"` will be parsed as
+    /// `Foo".bar` (note the preserved case and requiring two double quotes to represent
```

Comment:
```suggestion
    /// `Foo.bar` (note the preserved case and requiring two double quotes to represent
```

Reply:
It needs to contain the single double quote since it was escaped by prepending it with another double quote in the original, unless you mean to change the example?

Reply:
No, I think I was confused. Sorry about that